spigot/src/main/java/org/geysermc/floodgate/util/ClassNames.java

```diff
@@ -229,9 +252,50 @@ public class ClassNames {
         IS_FOLIA = ReflectionUtils.getClassSilently(
                 "io.papermc.paper.threadedregions.RegionizedServer"
         ) != null;
+
+        if (!IS_PRE_1_20_2) {
+            // PacketHandshakingInSetProtocol is now a record
+            // This means its fields are now private and final
+            // We therefore must use reflection to obtain the constructor
+            CLIENT_INTENT = getClassOrFallback(
+                    "net.minecraft.network.protocol.handshake.ClientIntent",
+                    nmsPackage + "ClientIntent"
+            );
+            checkNotNull(CLIENT_INTENT, "Client intent enum");
+
+            HANDSHAKE_PACKET_CONSTRUCTOR = getConstructor(HANDSHAKE_PACKET, false, int.class,
+                    String.class, int.class, CLIENT_INTENT);
+            checkNotNull(HANDSHAKE_PACKET_CONSTRUCTOR, "Handshake packet constructor");
+
+            HANDSHAKE_PORT = getField(HANDSHAKE_PACKET, "a");
+            checkNotNull(HANDSHAKE_PORT, "Handshake port");
+            makeAccessible(HANDSHAKE_PORT);
+
+            HANDSHAKE_PROTOCOL = getField(HANDSHAKE_PACKET, "c");
+            checkNotNull(HANDSHAKE_PROTOCOL, "Handshake protocol");
+            makeAccessible(HANDSHAKE_PROTOCOL);
+
+            HANDSHAKE_INTENTION = getFieldOfType(HANDSHAKE_PACKET, CLIENT_INTENT);
+            checkNotNull(HANDSHAKE_INTENTION, "Handshake intention");
+            makeAccessible(HANDSHAKE_INTENTION);
+        } else {
+            CLIENT_INTENT = null;
+            HANDSHAKE_PACKET_CONSTRUCTOR = null;
+            HANDSHAKE_PORT = null;
+            HANDSHAKE_PROTOCOL = null;
+            HANDSHAKE_INTENTION = null;
+        }
     }
 
     private static <T> T checkNotNull(@CheckForNull T toCheck, @CheckForNull String objectName) {
         return Preconditions.checkNotNull(toCheck, objectName + " cannot be null");
     }
+
+    // Ensure one of two is not null
+    private static <T> T checkNotNull(@CheckForNull T toCheck, @CheckForNull T toCheck2,
+                                      @CheckForNull String objectName,
+                                      @CheckForNull String objectName2) {
```

Comment:
If the limit is reached just do one argument per line like the SpigotCommandUtil constructor for example

Reply:
https://github.com/GeyserMC/Floodgate/pull/449/commits/2c52392451dc6af607b9b2194f6f460ecaff96a8#diff-c9d452f82010aef19e8cd407cb82aa21dff9a8fc01101c649ae2acb6e833b7e5R295-R300